ui/src/features/Dashboard/components/Nav/components/WorkspaceNavigation.tsx

```diff
@@ -9,22 +9,23 @@ import {
   DropdownMenuItem,
   DropdownMenuTrigger,
 } from "@flow/components";
+import { useGetWorkspaceQuery } from "@flow/lib/gql";
 import { cn } from "@flow/lib/utils";
-import { workspaces } from "@flow/mock_data/workspaceData";
-import { useCurrentProject, useCurrentWorkspace } from "@flow/stores";
+import { useCurrentWorkspace } from "@flow/stores";
 import { Workspace } from "@flow/types";
 
 const WorkspaceNavigation: React.FC = () => {
-  const [currentWorkspace, setCurrentWorkspace] = useCurrentWorkspace();
-  const [, setCurrentProject] = useCurrentProject();
-  const navigate = useNavigate({ from: "/workspace/$workspaceId" });
+  const [currentWorkspace, _] = useCurrentWorkspace();
+  const navigate = useNavigate();
 
   const handleWorkspaceChange = (workspace: Workspace) => {
-    setCurrentProject(undefined);
-    setCurrentWorkspace(workspace);
     navigate({ to: `/workspace/${workspace.id}` });
   };
 
+  const { data } = useGetWorkspaceQuery();
+
+  const workspaces = data?.me?.workspaces;
```

Comment:
I think querying and setting workspaces (or others) should hopefully only be in one place per query per page.
This might not be possible later on, but I feel like for objects that won't change often, like workspace or user, we should fetch this data as high up as possible and assign it to the global state. Then, use the global state in child components like WorkspaceNavigation. That way we have state initialization (query + global state set) in one place, which simplifies and synchronizes everything.
Make sense?